### PR TITLE
Add blaze size preference #125

### PR DESCRIPTION
**Pull request by aldebaran30701**
### Description of the Change
This pull request adds an extra UI tab to the preferences menu located in Setup -> Options -> Constellation -> Graph. The tab is responsible for global graph preferences upon loading a graph with no saved preferences. The issue was targeted at assigning a default blaze preference, but was extended to opacity also, due to ease of implementation and usability. Two slide controls have been added and control the values between 10 and 100 for both properties. 
The UI changes follow a similar layout structure as other preference tabs already utilized (Application). 
The previously assigned default values have been replaced with look-ups to globally stored preferences. In the case where no globally stored preference file exists, a default value still exists at the highest point in the lookup process (GraphPreferenceKeys).

The flow of logic is as follows:
 - Application start
 - Read of value from Preferences file (Default Value if no file exists)
 - Preferences Graph slider changed
 - Value written to Preferences file
 - New graph button clicked
 - New graph created with value stored in point 2
 
The behaviour above is not ideal, although allows the user to set a default value and have it refresh whenever the application restarts.

### Alternate Designs
The few options trialed to fix this issue were to utilize an accessor and mutator for fetching values from GraphPreferenceKeys, as opposed to requests to nbPreferences.forModule(...). Review of ways to "Refresh" or "Update" the SchemaFactory seemed not feasible, and were not implemented. Flush() - ing and Sync() - ing were both tested with the nbPreferences class in an attempt to ensure the values were written. The issue seemed to reflect the storage of the variable within application start and the lack of re-evaluation at run time. 

### Why Should This Be In Core?
This pull request adds functionality to allow a default value to be altered. Which, when creating a new graph, utilizes the new default value. It adds to the UX of the application and avoids making the user reset the blaze size whenever they create a new graph.

### Benefits
Extensibility of the UI for Graph property storage has been enhanced, which will support further development. The fix also, as mentioned above: reduces the amount of time that the user spends resetting blaze sizes when a new graph is made.

### Possible Drawbacks
Drawbacks of this implementation include the necessity to restart the application before changes become active. This is due to VisualConcept.java SchemaAttribute variables being assigned at application start. The current architecture does not allow for easy refactoring nor a proper workaround as many obstacles are in the way. Obstacles such as final declarations, unmodifiableCollections and statically typed classes make changes or updates to variables during run time difficult.

### Verification Process
All Tests were ran to verify regression, and human interaction with the tool was used as a basis for testing. 
 - launch constellation -> open new graph -> check blaze size = should be default value
 - launch constellation -> open preferences -> change preference values -> verify changed value in preferences.properties file = should be same value on slider
 - launch constellation -> change preferences -> open new graph = graph uses old preferences value (should use new preferences value, reason listed above)
 - launch constellation -> change preference values -> close application -> open application -> new graph -> check blaze values = should be saved new default value
 - launch constellation -> change blaze values in slider -> save file -> close file -> open file -> check blaze values = should be same as saved blaze size in file
 - launch constellation ->open multiple graphs -> change blaze slider on each -> verify they keep state and are different = states should be different

### Applicable Issues
As mentioned above, the minor issue of application restart needed for new preference value to become effective.

### CODE REVIEW

File: CorePreferences/src/au/gov/asd/tac/constellation/preferences/GraphOptionsPanelController.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/*
```
Comment (arcturus2) [inline CorePreferences/src/au/gov/asd/tac/constellation/preferences/GraphOptionsPanelController.java]: Please add the copyright header to this file too. If you haven't done so already have a look at https://github.com/constellation-app/constellation/blob/master/CONTRIBUTING.md#pull-requests

File: CorePreferences/src/au/gov/asd/tac/constellation/preferences/GraphPreferenceKeys.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    public static final String CHART_DISPLAY = "chartDisplay";
+    public static final String CHART_DISPLAY_CONSTELLATION = "constellation";
+    public static final String CHART_DISPLAY_BROWSER = "browser";
+    public static final String CHART_DISPLAY_DEFAULT = CHART_DISPLAY_CONSTELLATION;
+    /**
+     * Scripting.
+     */
+    public static final String DEFAULT_TEMPLATE = "defaultTemplate";
+    public static final String DEFAULT_TEMPLATE_DEFAULT = null;
```
Comment (arcturus2) [inline CorePreferences/src/au/gov/asd/tac/constellation/preferences/GraphPreferenceKeys.java]: Do we need these constants here?

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/concept/VisualConcept.java [hunk 3]
```diff
@@ -82,12 +86,12 @@ public String getName() {
                 .build();
         public static final SchemaAttribute BLAZE_OPACITY = new SchemaAttribute.Builder(GraphElementType.GRAPH, FloatAttributeDescription.ATTRIBUTE_NAME, "blaze_opacity")
                 .setDescription("The opacity of blazes on the graph")
-                .setDefaultValue(1.0)
+                .setDefaultValue(((float) ((float)(PREFERENCES.getInt(GraphPreferenceKeys.BLAZE_OPACITY, GraphPreferenceKeys.BLAZE_OPACITY_DEFAULT)/100.0))))
```
Comment (arcturus2) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/concept/VisualConcept.java]: What's the reason behind the outside `float` cast? If NetBeans throws a warning perhaps try changing the division to `100f`. I don't have your code checked out so apologies if I've missed something obvious.

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/display/GraphVisualAccess.java [hunk 4]
```diff
@@ -23,6 +23,7 @@
 import au.gov.asd.tac.constellation.graph.utilities.ConnectionMode;
 import au.gov.asd.tac.constellation.graph.visual.color.ColorAttributeDescription;
 import au.gov.asd.tac.constellation.graph.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.preferences.GraphPreferenceKeys;
```
Comment (arcturus2) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/display/GraphVisualAccess.java]: Is this an unused import?

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/display/GraphVisualAccess.java [hunk 5]
```diff
@@ -42,6 +43,7 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import org.openide.util.NbPreferences;
```
Comment (arcturus2) [inline CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/display/GraphVisualAccess.java]: Is this an unused import too?

File: CorePreferences/src/au/gov/asd/tac/constellation/preferences/GraphOptionsPanelController.java (hunk 1 above)
Comment (aldebaran30701) [reply CorePreferences/src/au/gov/asd/tac/constellation/preferences/GraphOptionsPanelController.java]: See latest commit "Resolve PR changes" where the license info is added

File: CorePreferences/src/au/gov/asd/tac/constellation/preferences/GraphPreferenceKeys.java (hunk 2 above)
Comment (aldebaran30701) [reply CorePreferences/src/au/gov/asd/tac/constellation/preferences/GraphPreferenceKeys.java]: See latest commit "Resolve PR changes" where I remove the unused constants

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/concept/VisualConcept.java (hunk 3 above)
Comment (aldebaran30701) [reply CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/concept/VisualConcept.java]: See latest commit "Resolve PR changes" where I change from double casted floats, to division by a float value

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/display/GraphVisualAccess.java (hunk 4 above)
Comment (aldebaran30701) [reply CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/display/GraphVisualAccess.java]: See latest commit "Resolve PR changes" Yes, It is.

File: CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/display/GraphVisualAccess.java (hunk 5 above)
Comment (aldebaran30701) [reply CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/display/GraphVisualAccess.java]: See latest commit "Resolve PR changes" Yes, It is.

**Comment by arcturus2** [comment]
Nice work @aldebaran30701. This is what I'm seeing when I look at it.

image

A few comments if I may:
* Given changes take affect after you restart Constellation, can you add a text label that says something to the effect of "Changes take effect after restarting Constellation" ?
* I'm wondering if we can remove the words "utilised on loading of a new graph" for the 2 options?
* Nitpicking...the Graph canvas size is bigger than it needs to be which is why there is a vertical scroll bar appearing. Please try to reduce it and don't hate me too much for this minor thing. You can have a look at the Proxy or Developer tabs and you'll see there isn't a scroll bar given there isn't enough content to overflow.

**Comment by arcturus2** [comment]
Perfect!

image

**Pull request merged**